client/client.go

```diff
@@ -266,10 +268,22 @@ func NewClientWithSocket(socket string) (client Client, err error) {
 		return nil, errors.New("nil socket")
 	}
 	logger.Printf("dapr client initializing for: %s", socket)
-	addr := fmt.Sprintf("unix://%s", socket)
-	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	timeoutSeconds, err := getClientTimeoutSeconds()
+	if err != nil {
+		return nil, err
+	}
+	addr := "unix://" + socket
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeoutSeconds)*time.Second)
```

Comment:
Working on this, I found a bug here. There's no need to save the context because it's only used by the `DialContext` method, and it's expired after the timeout anyways. So, I've removed saving the context in the object